addons/html_builder/static/src/builder/plugins/overlay_buttons/overlay_buttons_plugin.js

```diff
@@ -0,0 +1,175 @@
+import { Plugin } from "@html_editor/plugin";
+import { reactive } from "@odoo/owl";
+import { throttleForAnimation } from "@web/core/utils/timing";
+import { getScrollingElement, getScrollingTarget } from "@web/core/utils/scrolling";
+import { OverlayButtons } from "./overlay_buttons";
+import {
+    sizingY,
+    sizingX,
+    sizingGrid,
+} from "@html_builder/builder/plugins/builder_overlay/builder_overlay";
+
+// Moving with the arrows.
+const moveUpOrDown = {
+    selector: [
+        "section",
+        ".s_accordion .accordion-item",
+        ".s_showcase .row .row:not(.s_col_no_resize) > div",
+        ".s_hr",
+        // In snippets files
+        ".s_pricelist_boxed_item",
+        ".s_pricelist_cafe_item",
+        ".s_product_catalog_dish",
+        ".s_timeline_list_row",
+        ".s_timeline_row",
+    ].join(", "),
+};
+
+const moveLeftOrRight = {
+    selector: [
+        ".row:not(.s_col_no_resize) > div",
+        ".nav-item",
+        ".s_timeline_card", // timeline TODO custom function, other plugin ?
+    ].join(", "),
+    exclude: ".s_showcase .row .row > div",
+};
+
+export class OverlayButtonsPlugin extends Plugin {
+    static id = "overlayButtons";
+    static dependencies = ["selection", "overlay", "history"];
+    resources = {
+        step_added_handlers: this.refreshButtons.bind(this),
+        change_current_options_containers_listeners: this.addOverlayButtons.bind(this),
+    };
+
+    setup() {
+        this.overlay = this.dependencies.overlay.createOverlay(OverlayButtons, {
+            positionOptions: {
+                position: "top-middle",
+                onPositioned: (overlayEl, position) => {
+                    if (position.top < 0) {
+                        const targetRect = this.target.getBoundingClientRect();
+                        const newTop = targetRect.bottom + 15;
+                        position.top = newTop;
+                        overlayEl.style.top = `${newTop}px`;
+                    }
+                    return;
+                },
+                margin: 15,
+                flip: false,
+            },
+            closeOnPointerdown: false,
+        });
+        this.target = null;
+        this.state = reactive({
+            isVisible: true,
+            buttons: [],
+        });
+
+        this.resizeObserver = new ResizeObserver(() => {
+            this.overlay.updatePosition();
+        });
+
+        // TODO duplicate of builderOverlay => extract somewhere
+        // Recompute the buttons when the window is resized.
+        this.refresh = throttleForAnimation(this.refreshButtons.bind(this));
+        this.addDomListener(window, "resize", this.refresh);
+
+        // On keydown, hide the buttons and then show them again when the mouse
+        // moves.
+        const onMouseMoveOrDown = throttleForAnimation((ev) => {
+            this.toggleVisibility(true);
```

Comment:
Why not close it and open it again when you want to make it visible ?

Reply:
I find it more correct to just hide it, as the target didn't change during that time :slightly_smiling_face: